src/main/java/com/denizenscript/denizencore/scripts/commands/core/ImageCommand.java

```diff
@@ -0,0 +1,182 @@
+package com.denizenscript.denizencore.scripts.commands.core;
+
+import com.denizenscript.denizencore.DenizenCore;
+import com.denizenscript.denizencore.exceptions.InvalidArgumentsRuntimeException;
+import com.denizenscript.denizencore.objects.core.ImageTag;
+import com.denizenscript.denizencore.scripts.ScriptEntry;
+import com.denizenscript.denizencore.scripts.commands.AbstractCommand;
+import com.denizenscript.denizencore.scripts.commands.Holdable;
+import com.denizenscript.denizencore.scripts.commands.file.FileReadCommand;
+import com.denizenscript.denizencore.scripts.commands.file.FileWriteCommand;
+import com.denizenscript.denizencore.scripts.commands.generator.ArgDefaultNull;
+import com.denizenscript.denizencore.scripts.commands.generator.ArgDefaultText;
+import com.denizenscript.denizencore.scripts.commands.generator.ArgName;
+import com.denizenscript.denizencore.scripts.commands.generator.ArgPrefixed;
+import com.denizenscript.denizencore.utilities.CoreUtilities;
+import com.denizenscript.denizencore.utilities.debugging.Debug;
+
+import javax.imageio.ImageIO;
+import java.awt.image.BufferedImage;
+import java.io.File;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+public class ImageCommand extends AbstractCommand implements Holdable {
+
+    // <--[command]
+    // @Name Image
+    // @Syntax image [id:<id>] [load [image:<image>/path:<path>]]/[save [path:<path>] [format:<format>]]/[unload]
+    // @Required 2
+    // @Maximum 4
+    // @Short Loads, saves, and unloads images.
+    // @Group image
+    //
+    // @Description
+    // Loads, saves, and unloads images.
+    //
+    // With "load", specify either a file path to read from or an image object to load.
+    // With "save", specify a file path to save the image to and a format to save the image in (e.g. "png", "jpg", "bmp"...), defaults to "png".
+    // For both of these the starting path is "plugins/Denizen".
+    // Use waitable syntax ("- ~image") when loading or saving from a file to avoid locking up the server during file IO, refer to <@link language ~waitable>.
+    //
+    // All uses of the image command must include the "id:" argument. This is any arbitrary name, as plaintext or from a tag,
+    // to uniquely and globally identify the image object in memory. This ID can only be used by one image object at a time.
+    // IDs are stored when "load" is used, and only removed when "unload" is used.
+    //
+    // @Tags
+    // None
+    //
+    // @Usage
+    // Use to load an image from a file into memory under the id "image".
+    // - ~image id:image load path:data/image.png
+    //
+    // @Usage
+    // Use to load an image object from a definition, draw on it, then store it back into the definition.
```

Comment:
This example should be updated per image-identified-by-id change

Reply:
Repaired